### PR TITLE
RFR: Beginning of POST book_from_csv

### LINKED ISSUE #119: Parse CSV into Book model
**Describe the solution you'd like**
A BE endpoint that will support parsing a CSV file and creating a Book model in the backend.

### PR DESCRIPTION
**Pull request by micleo2**
:rocket: Ready

Related issues: closes #119

### CODE REVIEW

File: backend/api/views/book.py [hunk 1]
```diff
@@ -19,6 +21,44 @@ def invalid_book_data(user_data):
...
+@book.route("/book_from_csv", methods=["POST"])
+def create_book_from_csv():
+    uploaded_csv = request.files["File"]
+    if not uploaded_csv:
+        return create_response(message="Missing CSV file", status=409)
+
+    stream = io.StringIO(uploaded_csv.stream.read().decode("UTF8"), newline=None)
+    parsed_data = csv.reader(stream)
+    header = next(parsed_data)
+    header = [col.replace(" ", "_") for col in next(parsed_data)]
```
Comment (Davidw1339) [inline backend/api/views/book.py]: good check/edit :checkered_flag:

File: backend/api/views/book.py [hunk 2]
```diff
@@ -19,6 +21,44 @@ def invalid_book_data(user_data):
...
+    for row in parsed_data:
+        row_dict = {
+            key.lower(): value for key, value in zip(header, row)
+        }  # resilient to future changes of column positions
```
Comment (Davidw1339) [inline backend/api/views/book.py]: nice! :+1:

File: backend/api/views/book.py [hunk 3]
```diff
@@ -19,6 +21,44 @@ def invalid_book_data(user_data):
...
+        if invalid_book_data(row_dict):
+            return create_response(
+                message=f"Invalid book data {row}",
+                status=409,
+                data={"status": "failure"},
```
Comment (Davidw1339) [inline backend/api/views/book.py]: Nit: very minor but is there a reason to have data dict here, but not in the previous create_response: line 28 or 58

**Comment by XuanLi-Leong** [comment]
Could you create a `csv` template that we can let admins download from our page? It'd be great to populate it with some sample data too for extra clarity on how to use it.

**Comment by Davidw1339** [comment]
> Could you create a `csv` template that we can let admins download from our page? It'd be great to populate it with some sample data too for extra clarity on how to use it.

Sure thing, Michael was using one to test the adding books endpoint

**Pull request merged**